src/vs/platform/quickinput/browser/tree/quickInputTreeController.ts

```diff
@@ -21,6 +22,12 @@ import { QuickInputTreeSorter } from './quickInputTreeSorter.js';
 
 const $ = dom.$;
 
+class QuickInputTreeIdentityProvider implements IIdentityProvider<IQuickTreeItem> {
+	getId(element: IQuickTreeItem): { toString(): string } {
+		return { toString: () => element.id ?? '' };
```

Comment:
id is optional on element.

https://github.com/microsoft/vscode/blob/782c46dde9a1572b41293ad25042b0306bcf5333/src/vs/platform/quickinput/common/quickInput.ts#L34

we should have a better fallback here than `''`